### PR TITLE
completely refactor README & parts of docs

### PR DESCRIPTION
**Pull request by georglauterbach**
# Description

I removed many parts of the `README` and placed them inside the docs. This is done to ensure we can properly version all of out documentation. I believe that users should generally **always use our documentation**, not a mixture of `README` + docs, to configure and check their mail server. This makes maintenance easier for maintainers, and changes are local to the version. Previously, we had to keep the `README` and docs in sync, which is undesirable.

I took many parts of the `README` and copy-pasted it into the docs. Other parts were slightly adjusted to match the context. The landing page of our docs was more heavily refactored and updated to display more information.

I also added notes on the versioning of the docs!

There is actually more work to do in cleaning up parts of the docs (a section in the examples chapter needs overhaul, etc.). But I did want to make this PR bigger - it is big enough.

---

@polarathene please, by all means, go ahead and apply suggested changes immediately without waiting for me. I'd only just apply them anyway :) This way we can make reviewing faster. I hope you think the changes a worthwhile.

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)

### CODE REVIEW

**Comment by georglauterbach** [comment]
@wernerfred I you have the time, please provide me with some feedback and whether you think this is a good idea :)

File: docs/content/examples/tutorials/basic-installation.md [hunk 1]
```diff
@@ -2,9 +2,107 @@
...
+    restart: always
+```
+
+## A Detailed Example
+
+!!! note
+
+    This is a community contributed guide. Please let us know via a Github Issue if you're having any difficulty following the guide so that we can update it.
+
+    This section probably needs an overhaul!
```
Comment (wernerfred) [inline docs/content/examples/tutorials/basic-installation.md:105]: laughed on that 😄 sure you want to keep it?

**Comment by wernerfred** [comment]
> @wernerfred I you have the time, please provide me with some feedback and whether you think this is a good idea :)

it absolutely is!

> Previously, we had to keep the `README` and docs in sync, which is undesirable.

Finally, we should have done that much earlier! 

> The landing page of our docs was more heavily refactored and updated to display more information.

It reads fantastic now imho. Much cleaner and futureproof!

Thanks a lot!

**Review by wernerfred — APPROVED** [review]
🚀

File: docs/content/faq.md [hunk 2]
```diff
@@ -4,49 +4,91 @@ title: 'FAQ'
...
+docker-compose pull
+docker-compose down
+docker-compose up -d mailserver
+```
+
+You should see the new version number on startup, for example: `[   INF   ]  Welcome to docker-mailserver 11.3.1`. And you're done! Don't forget to have a look at the remaining functions of the `setup.sh` script with `./setup.sh help`.
+
+### Which operating systems are supported?
+
+We are currently providing support for Linux. Windows is _not_ supported and is known to cause problems. Similarly, macOS is _not officially_ supported - but you may get it to work there. In the end, Linux should be your preferred operating system for this image, especially when using this mail-server in production.
```
Comment (polarathene) [inline docs/content/faq.md:33]: Revised:

```suggestion
- Linux is officially supported.
- Windows and macOS are _not_ supported and users and have reported various issues running the image on these hosts.

As you'll realistically be deploying to production on a Linux host, if you are on Windows or macOS and want to run the image locally first, it's advised to do so via a VM guest running Linux if you have issues running DMS on your host system.
```

File: docs/content/faq.md [hunk 3]
```diff
@@ -4,49 +4,91 @@ title: 'FAQ'
...
+### How do I update DMS?
+
+**Make sure to read the [CHANGELOG](https://github.com/docker-mailserver/docker-mailserver/blob/master/CHANGELOG.md)** before updating to new versions, to be prepared for possible breaking changes.
+
+Then, run the following commands:
+
+``` BASH
+docker-compose pull
+docker-compose down
+docker-compose up -d mailserver
```
Comment (polarathene) [inline docs/content/faq.md:26]: Prior `down` command didn't specify a service, if there's more than one, you probably want to bring them back up too? User should know either way.

```suggestion
docker-compose up -d
```

File: README.md [hunk 4]
```diff
@@ -9,312 +9,40 @@
...
-8. [Release Notes](./CHANGELOG.md)
+7. [Updating](https://docker-mailserver.github.io/docker-mailserver/edge/faq/#updating)
 
-## Included Services
+## :package: Included Services
 
-- [Postfix](http://www.postfix.org) with SMTP or LDAP auth
-- [Dovecot](https://www.dovecot.org) for SASL, IMAP (or POP3), with LDAP Auth, Sieve and [quotas](https://docker-mailserver.github.io/docker-mailserver/edge/config/user-management/accounts#notes)
+- [Postfix](http://www.postfix.org) with SMTP or LDAP authentication and support for [extension delimiters](http://www.postfix.org/postconf.5.html#recipient_delimiter)
+- [Dovecot](https://www.dovecot.org) with SASL, IMAP or POP3, LDAP, [basic Sieve support](https://docker-mailserver.github.io/docker-mailserver/edge/config/advanced/mail-sieve) and [quotas](https://docker-mailserver.github.io/docker-mailserver/edge/config/user-management/accounts#notes)
```
Comment (polarathene) [inline README.md:37]:
```suggestion
- [Dovecot](https://www.dovecot.org) with SASL, IMAP, POP3, LDAP, [basic Sieve support](https://docker-mailserver.github.io/docker-mailserver/edge/config/advanced/mail-sieve) and [quotas](https://docker-mailserver.github.io/docker-mailserver/edge/config/user-management/accounts#notes)
```

File: README.md [hunk 5]
```diff
@@ -9,312 +9,40 @@
...
-      - SASLAUTHD_LDAP_SEARCH_BASE=ou=people,dc=localhost,dc=localdomain
-      - SASLAUTHD_LDAP_FILTER=(&(objectClass=PostfixBookMailAccount)(uniqueIdentifier=%U))
-      - POSTMASTER_ADDRESS=postmaster@localhost.localdomain
-      - POSTFIX_MESSAGE_SIZE_LIMIT=100000000
-    cap_add:
-      - NET_ADMIN
-    restart: always
-```
+- support for [LetsEncrypt](https://letsencrypt.org/), manual and self-signed certificates
+- a [setup script](https://docker-mailserver.github.io/docker-mailserver/edge/config/setup.sh) for easy configuration and maintenance
```
Comment (polarathene) [inline README.md:47]: Consistent captilization at start of line like other bullet points:

```suggestion
- Support for [LetsEncrypt](https://letsencrypt.org/), manual and self-signed certificates
- A [setup script](https://docker-mailserver.github.io/docker-mailserver/edge/config/setup.sh) for easy configuration and maintenance
```

File: README.md [hunk 6]
```diff
@@ -9,312 +9,40 @@
...
-7. [Documentation][documentation::web]
-8. [Release Notes](./CHANGELOG.md)
+7. [Updating](https://docker-mailserver.github.io/docker-mailserver/edge/faq/#updating)
 
-## Included Services
+## :package: Included Services
 
-- [Postfix](http://www.postfix.org) with SMTP or LDAP auth
-- [Dovecot](https://www.dovecot.org) for SASL, IMAP (or POP3), with LDAP Auth, Sieve and [quotas](https://docker-mailserver.github.io/docker-mailserver/edge/config/user-management/accounts#notes)
+- [Postfix](http://www.postfix.org) with SMTP or LDAP authentication and support for [extension delimiters](http://www.postfix.org/postconf.5.html#recipient_delimiter)
```
Comment (polarathene) [inline README.md:36]:
```suggestion
- [Postfix](http://www.postfix.org) with SMTP or LDAP authentication and support for [extension delimiters](http://www.postfix.org/postconf.5.html#recipient_delimiter) (_mail to `you+extension@example.com` delivered to `you@example.com`_)
```

File: docs/content/examples/tutorials/basic-installation.md [hunk 7]
```diff
@@ -2,9 +2,107 @@
...
+```
+
+## A Basic LDAP Setup
+
+**Note** There are currently no LDAP maintainers. If you encounter issues, please raise them in the issue tracker, but be aware that the core maintainers team will most likely not be able to help you. **We would appreciate and we encourage everyone to actively participate in maintaining LDAP-related code by becoming a maintainer!**
+
+``` YAML
+version: '3.8'
+
+services:
```
Comment (polarathene) [inline docs/content/examples/tutorials/basic-installation.md:49]:
```suggestion
services:
```

File: docs/content/examples/tutorials/basic-installation.md [hunk 8]
```diff
@@ -2,9 +2,107 @@
...
-
-This is a community contributed guide. Please let us know via a Github Issue if you're having any difficulty following the guide so that we can update it.
+## A Basic Example With Relevant Environmental Variables
+
+This example provides you only with a basic example of what a minimal setup could look like. We **strongly recommend** that you go through the configuration file yourself and adjust everything to your needs. The default [docker-compose.yml](https://github.com/docker-mailserver/docker-mailserver/blob/master/docker-compose.yml) can be used for the purpose out-of-the-box, see the [_Usage_ chapter](../../usage.md).
+
+``` YAML
+version: '3.8'
+
+services:
```
Comment (polarathene) [inline docs/content/examples/tutorials/basic-installation.md:12]: No version needed.

```suggestion
services:
```

File: docs/content/examples/tutorials/basic-installation.md [hunk 9]
```diff
@@ -2,9 +2,107 @@
...
+**Note** There are currently no LDAP maintainers. If you encounter issues, please raise them in the issue tracker, but be aware that the core maintainers team will most likely not be able to help you. **We would appreciate and we encourage everyone to actively participate in maintaining LDAP-related code by becoming a maintainer!**
+
+``` YAML
+version: '3.8'
+
+services:
+  mailserver:
+    image: docker.io/mailserver/docker-mailserver:latest
+    container_name: mailserver
+    # Provide the FQDN of your mail server her
```
Comment (polarathene) [inline docs/content/examples/tutorials/basic-installation.md:53]:
```suggestion
    # Provide the FQDN of your mail server here (Your DNS MX record should point to this value)
```

File: docs/content/examples/tutorials/basic-installation.md [hunk 10]
```diff
@@ -2,9 +2,107 @@
...
+    hostname: mail.example.com
+    ports:
+      - "25:25"
+      - "143:143"
```
Comment (polarathene) [inline docs/content/examples/tutorials/basic-installation.md:57]:
```suggestion
```

File: docs/content/examples/tutorials/basic-installation.md [hunk 11]
```diff
@@ -29,38 +127,30 @@ In this setup `docker-mailserver` is not intended to receive email externally, s
...
               - ./docker-data/dms/mail-state/:/var/mail-state/
               - ./docker-data/dms/mail-logs/:/var/log/mail/
               - ./docker-data/dms/config/:/tmp/docker-mailserver/
-              # The "from" path will vary based on where your certs are locally:
-              - ./docker-data/nginx-proxy/certs/:/etc/letsencrypt/
               - /etc/localtime:/etc/localtime:ro
             environment:
+              - ENABLE_SPAMASSASSIN=1
+              - SPAMASSASSIN_SPAM_TO_INBOX=1
+              - ENABLE_CLAMAV=1
```
Comment (polarathene) [inline docs/content/examples/tutorials/basic-installation.md:146]:
```suggestion
```

File: docs/content/examples/tutorials/basic-installation.md [hunk 12]
```diff
@@ -29,38 +127,30 @@ In this setup `docker-mailserver` is not intended to receive email externally, s
...
               - ENABLE_FAIL2BAN=1
-              # Using letsencrypt for SSL/TLS certificates
-              - SSL_TYPE=letsencrypt
-              # Allow sending emails from other docker containers
-              # Beware creating an Open Relay: https://docker-mailserver.github.io/docker-mailserver/edge/config/environment/#permit_docker
-              - PERMIT_DOCKER=network
-              # All env below are default settings:
+              - ENABLE_POSTGREY=1
+              - ENABLE_SASLAUTHD=0
```
Comment (polarathene) [inline docs/content/examples/tutorials/basic-installation.md:149]: Bringing back original ENV:

```suggestion
              # Using letsencrypt for SSL/TLS certificates
              - SSL_TYPE=letsencrypt
              # Allow sending emails from other docker containers
              # Beware creating an Open Relay: https://docker-mailserver.github.io/docker-mailserver/edge/config/environment/#permit_docker
              - PERMIT_DOCKER=network
              # You may want to enable this: https://docker-mailserver.github.io/docker-mailserver/edge/config/environment/#spoof_protection
              # See step 8 below, which demonstrates setup with enabled/disabled SPOOF_PROTECTION:
              - SPOOF_PROTECTION=0
```

File: docs/content/examples/tutorials/basic-installation.md [hunk 13]
```diff
@@ -29,38 +127,30 @@ In this setup `docker-mailserver` is not intended to receive email externally, s
...
               - ONE_DIR=1
-              - ENABLE_POSTGREY=0
-              - ENABLE_CLAMAV=0
-              - ENABLE_SPAMASSASSIN=0
-              # You may want to enable this: https://docker-mailserver.github.io/docker-mailserver/edge/config/environment/#spoof_protection
-              # See step 8 below, which demonstrates setup with enabled/disabled SPOOF_PROTECTION:
-              - SPOOF_PROTECTION=0
             cap_add:
-              - NET_ADMIN # For Fail2Ban to work
+              - NET_ADMIN
```
Comment (polarathene) [inline docs/content/examples/tutorials/basic-installation.md:152]:
```suggestion
              - NET_ADMIN # For Fail2Ban to work
```

File: docs/content/examples/tutorials/basic-installation.md (hunk 1 above)
Comment (polarathene) [inline docs/content/examples/tutorials/basic-installation.md:105]:
```suggestion
```

**Review by polarathene — CHANGES_REQUESTED** [review]
Great work tackling this :grinning: 

I'll apply my suggestions via the web UI.

File: docs/content/examples/tutorials/basic-installation.md [hunk 14]
```diff
@@ -29,38 +127,30 @@ In this setup `docker-mailserver` is not intended to receive email externally, s
...
+            restart: always
```
[the remark this answers is not in the document]
Comment (polarathene) [reply docs/content/examples/tutorials/basic-installation.md]: Resolved: Reverted config

File: docs/content/examples/tutorials/basic-installation.md (hunk 1 above)
Comment (polarathene) [reply docs/content/examples/tutorials/basic-installation.md]: Removed overhaul line. Guide is fine for now, could do with a bit of revision but good enough.

**Review by polarathene — DISMISSED** [review]
LGTM :+1:

**Comment by georglauterbach** [comment]
I resolved the conflicts and will merge it now :)

File: docs/content/examples/tutorials/basic-installation.md (hunk 1 above)
Comment (georglauterbach) [reply docs/content/examples/tutorials/basic-installation.md]: Haha, this was supposed to be a comment for maintainers/me, but I see what you mean. @polarathene is correct in removing it :D

**Pull request merged**